feat: Added numerous trigonometric UDFs (#9213)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/scalar-functions.md

---

### **`ACOS`**

```sql title="Since: 0.27.0"
ACOS(col1)
```

> Returns the inverse (arc) cosine of `col1` in radians. Use [DEGREES](#degrees) to convert the output to degrees if necessary.

Reviewer: ```suggestion
Returns the inverse (arc) cosine of `col1`, in radians. Use the [DEGREES](#degrees) function to convert the output to degrees.
```

> Note that this function will return `NaN` for any input outside [-1, 1].

Reviewer: ```suggestion
This function returns `NaN` for any input outside [-1, 1].
```

---

### **`ASIN`**

```sql title="Since: 0.27.0"
ASIN(col1)
```

> Returns the inverse (arc) sine of `col1` in radians. Use [DEGREES](#degrees) to convert the output to degrees if necessary.

Reviewer: ```suggestion
Returns the inverse (arc) sine of `col1`, in radians. Use the [DEGREES](#degrees) function to convert the output to degrees.
```

> Note that this function will return `NaN` for any input outside [-1, 1].

Reviewer: ```suggestion
This function returns `NaN` for any input outside [-1, 1].
```

---

### **`ATAN`**

```sql title="Since: 0.27.0"
ATAN(col1)
```

> Returns the inverse (arc) tangent of `col1` in radians. Use [DEGREES](#degrees) to convert the output to degrees if necessary.

Reviewer: ```suggestion
Returns the inverse (arc) tangent of `col1`, in radians. Use the [DEGREES](#degrees) function to convert the output to degrees.
```

---

### **`ATAN2`**

```sql title="Since: 0.27.0"
ATAN2(y, x)
```

> Returns the inverse (arc) tangent of `y / x`. This is equivalent to the angle theta when Cartesian coordinates (x, y) are converted to polar coordinates (radius, theta). The returned value is in radians. Use [DEGREES](#degrees) to convert the output to degrees if necessary.

Reviewer: ```suggestion
Returns the inverse (arc) tangent of `y / x`. This is equivalent to the angle _theta_ when Cartesian coordinates (x, y) are converted to polar coordinates (_radius_, _theta_). The returned value is in radians. Use the [DEGREES](#degrees) function to convert the output to degrees.
```

> In cases where `x` is zero, and `y / x` is undefined, this function returns the approximate value of a multiple of π/2.

Reviewer: ```suggestion
If `x` is zero, `y / x` is undefined, and this function returns the approximate value of a multiple of _π/2_.
```

---

### **`COS`**

```sql title="Since: 0.27.0"
COS(col1)
```

> Returns the cosine of `col1`. `col1` is in radians. Use [RADIANS](#radians) to convert the input to radians if necessary.

Reviewer: ```suggestion
Returns the cosine of `col1`. `col1` is in radians. Use the [RADIANS](#radians) function to convert the input to radians, if necessary.
```

---

### **`PI`**

```sql title="Since: 0.27.0"
PI()
```

> Returns an approximate value of π.

Reviewer: ```suggestion
Returns an approximate value of _π_.
```

---

### **`SIN`**

```sql title="Since: 0.27.0"
SIN(col1)
```

> Returns the sine of `col1`. `col1` is in radians. Use [RADIANS](#radians) to convert the input to radians if necessary.

Reviewer: ```suggestion
Returns the sine of `col1`. `col1` is in radians. Use the [RADIANS](#radians) function to convert the input to radians, if necessary.
```

---

### **`TAN`**

```sql title="Since: 0.27.0"
TAN(col1)
```

> Returns the tangent of `col1`. `col1` is in radians. Use [RADIANS](#radians) to convert the input to radians if necessary.

Reviewer: ```suggestion
Returns the tangent of `col1`. `col1` is in radians. Use the [RADIANS](#radians) function to convert the input to radians, if necessary.
```

> This implementation returns a large value approaching positive or negative infinity near the asymptotes since π/2 and similar values cannot be represented exactly.

Reviewer: ```suggestion
This implementation returns a large value approaching positive or negative infinity near the asymptotes, because _π/2_ and similar values cannot be represented exactly.
```

---

### **`TANH`**

```sql title="Since: 0.27.0"
TANH(col1)
```

> Returns the hyperbolic tangent of `col1`. `col1` is in radians. Use [RADIANS](#radians) to convert the input to radians if necessary.

Reviewer: ```suggestion
Returns the hyperbolic tangent of `col1`. `col1` is in radians. Use the [RADIANS](#radians) function to convert the input to radians, if necessary.
```
